### ISSUE TITLE
Invalid notification timing

### ISSUE THREAD

**Issue by tas50** [chef/cookstyle#156]
### What category of cop is this?:

Please try to select as few as possible.

- [ ] Deprecation - A cop that alerts to a deprecation in the Chef Infra Client
- [ ] Correctness - A cop that alerts when incorrect coding behavior is being used
- [ ] Sharing - A cop that alerts to missing functionality necessary for sharing cookboooks with others
- [ ] Style - A cop that alerts to a style best practice
- [ ] Simplification - A cop that alerts to ways code can be simplified through refactoring
- [ ] Effortless - A cop that alerts on code that must be resolved to move to the Effortless pattern
- [ ] Policyfiles - A cop that alerts on code that must be resolved to move to the Policyfiles pattern

### Describe the new cop:
Notification timing can be controlled and set to immediate, or delayed until the end of the Chef run. This warning is shown when the timing specified is not recognised.

### What it would trigger on:

```ruby
template '/etc/foo' do
  notifies :restart, 'service[foo]', :imediate
end
```

### What it would autocorrect to if applicable

```ruby
template '/etc/foo' do
  notifies :restart, 'service[foo]', :imediately
end
```
